### PR TITLE
Feat/manager 6890: add attach data tab

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/notebook`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-6890
| License          | BSD 3-Clause

## Description

Add notebook attach_data tab into dashboard page.

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "pci_notebooks_tab_attach_data_page_description_link": "En savoir plus à propos du stockage de données.",
+  "pci_notebooks_tab_attach_data_page_volumes_column_name": "Nom",
+  "pci_notebooks_tab_attach_data_page_volumes_column_type": "Type",
+  "pci_notebooks_tab_attach_data_page_volumes_column_type_git": "Repositorie",
+  "pci_notebooks_tab_attach_data_page_volumes_column_type_swift": "Conteneur de données",
+  "pci_notebooks_tab_attach_data_page_volumes_column_mount_path": "Point de montage",
+  "pci_notebooks_tab_attach_data_page_volumes_column_permission": "Permission",
+  "pci_notebooks_tab_attach_data_page_volumes_column_permission_RO": "Lecture Seule",
+  "pci_notebooks_tab_attach_data_page_volumes_column_permission_RW": "Lecture & Ecriture"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json:13]: dont care about translation, it's a fake values.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -1,3 +1,4 @@
 {
-  "pci_notebooks_dashboard_info_tab_title": "Informations générales"
+  "pci_notebooks_dashboard_info_tab_title": "Informations générales",
+  "pci_notebooks_dashboard_attach_data_tab_title": "Attach data"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/translations/Messages_fr_FR.json:3]: same, don't care about content.
We still waiting a final content.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const NOTEBOOK_ATTACH_DATA_INFO_LINK = {
+  DEFAULT: 'https://docs.ovh.com/fr/logs-data-platform/kubernetes-fluent-bit/',
```
Comment (mohammed-zahaf) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js:2]: still waiting for the link

**Comment by mohammed-zahaf** [comment]
Hello @frenautvh, it's ready for the review.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+};
+
+export const NOTEBOOK_ATTACH_DATA_DATAGRID = {
+  MAX_ITEMS_PER_PAGE: 10,
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js:6]: Don't know if it's really useful, this constant is used only in one place.
Would make more sens to have a common pagination for all datagrids (but it's another topic).

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.component.js [hunk 5]
```diff
@@ -2,10 +2,12 @@ import template from './dashboard.html';
...
 export default {
   bindings: {
-    currentActiveLink: '<',
     notebook: '<',
-    generalInformationLink: '<',
     guideUrl: '<',
+    currentActiveLink: '<',
+    generalInformationLink: '<',
+    attachDataLink: '<',
+    test: '@',
```
Comment (cbourgois) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.component.js:10]: Useful ?

```suggestion
```

**Review by cbourgois — CHANGES_REQUESTED** [review]
Waiting some content and one `test` to remove, otherwise seems ok.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    $translate,
+    coreConfig,
+    CucCloudMessage,
+    CucRegionService,
+    NotebookService,
+  ) {
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.CucRegionService = CucRegionService;
+    this.NotebookService = NotebookService;
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js:18]: service doesn't seem needed

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                property="name"
+            >
+            </oui-datagrid-column>
+            <oui-datagrid-column
+                title=":: 'pci_notebooks_tab_attach_data_page_volumes_column_type' | translate"
+                sortable="true"
+                property="volumeType"
+            >
+                {{:: 'pci_notebooks_tab_attach_data_page_volumes_column_type_' +
+                $row.volumeType | translate}}
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html:52]:
```suggestion
                <span data-translate="{{'pci_notebooks_tab_attach_data_page_volumes_column_type_' +
                $row.volumeType}}"></span>
```

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
+{
+  "pci_notebooks_tab_attach_data_title": "Associer des données",
+  "pci_notebooks_tab_attach_data_page_title": "Attacher des conteneurs de données et un repository",
+  "pci_notebooks_tab_attach_data_page_description": "Si nécessaire, vous pouvez attacher des conteneurs Object Storage OVHcloud à votre notebook. Une fois attachés, ceux-ci seront temporairement chargés et mis en cache près de votre instance afin de réduire la latence et d’améliorer les performances. La bonne pratique est d’attacher un conteneur avec vos données entrantes et un autre avec vos données sortantes.",
+  "pci_notebooks_tab_attach_data_page_description_link": "En savoir plus sur le stockage de données.",
+  "pci_notebooks_tab_attach_data_page_volumes_column_name": "Nom",
+  "pci_notebooks_tab_attach_data_page_volumes_column_type": "Type",
+  "pci_notebooks_tab_attach_data_page_volumes_column_type_git": "Repositorie",
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json:8]:
```suggestion
  "pci_notebooks_tab_attach_data_page_volumes_column_type_git": "Repository",
```

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js (hunk 6 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.controller.js:18]: Fixed

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js (hunk 4 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js:6]: Moved into controller.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json (hunk 8 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/translations/Messages_fr_FR.json:8]: Fixed

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/translations/Messages_fr_FR.json (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/translations/Messages_fr_FR.json:3]: Fixed.

**Comment by marie-j** [comment]
Hello @mohammed-zahaf 
Could you please resolve existing conflicts ?
Thanks

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js (hunk 3 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.constants.js]: Link has been provided

**Comment by mohammed-zahaf** [comment]
> Hello @mohammed-zahaf
> Could you please resolve existing conflicts ?
> Thanks

Hello @marie-j, conflicts has been fixed.
Thanks

**Review by radireddy — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.routing.js [hunk 9]
```diff
@@ -9,8 +10,17 @@ export default /* @ngInject */ ($stateProvider) => {
...
         $transition$.params().notebookId,
-      notebook: /* @ngInject */ (notebookId, notebooks) =>
-        find(notebooks, { id: notebookId }),
+      notebook: /* @ngInject */ (
+        notebookId,
+        notebooks,
+        coreConfig,
+        CucRegionService,
+      ) =>
+        new Notebook(
```
Comment (ganeshkumar1989) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.routing.js:19]: Why do you need to create a new object here? If is it `coreConfig` and `CucRegionService`, cant we pass it in the top level route itself?
And I don't feel it is right to pass `coreConfig` and `CucRegionService` (which are angular services) into a native class...

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+                    class="oui-icon oui-icon-external_link"
+                    aria-hidden="true"
+                ></span>
+            </a>
+        </div>
+
+        <!--Table-->
+        <oui-datagrid
+            rows="$ctrl.notebook.spec.volumes"
+            page-size="$ctrl.DATAGRID_MAX_ITEMS_PER_PAGE"
```
Comment (ganeshkumar1989) [inline packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html:38]: `page-size="::$ctrl.DATAGRID_MAX_ITEMS_PER_PAGE"`

**Comment by ganeshkumar1989** [comment]
Please rebase with parent branch to resolve conflicts :)

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.routing.js (hunk 9 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.routing.js]: Now i got the notebook via the list.

File: packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html (hunk 10 above)
Comment (mohammed-zahaf) [reply packages/manager/modules/pci/src/projects/project/notebooks/dashboard/attach-data/attach-data.html:38]: fixed

**Comment by mohammed-zahaf** [comment]
Hello @ganeshkumar1989, i fixed all remarks.
Can you take a look again please, thanks

**Review by ganeshkumar1989 — APPROVED** [review]

**Pull request merged**